src/redisraft.c

```diff
@@ -1000,9 +1000,15 @@ static int cmdRaftSnapshot(RedisModuleCtx *ctx, RedisModuleString **argv, int ar
 
     rr->snapshotreq_received++;
 
+    /* raft_recv_snapshot() might trigger loading the received RDB file.
+     * While loading, Redis may process incoming messages to reply -LOADING.
+     * So, we must block the leader's connection to prevent processing further
+     * messages, as we haven't replied to the current command yet. */
+    RedisModuleBlockedClient *c = RedisModule_BlockClient(ctx, NULL, 0, 0, 0);
```

Comment:
I'm confused by the purpose of this.  I thought BlockClient was blocking the client (i.e. essentially allowing it to be backgrounded), not the server.

Reply:
`rdbLoad()` is synchronous but inside that function, it still processes messages from the network. It hits here: https://github.com/redis/redis/blob/599e59ebc57283f52c60a8de56ec5f44d053109a/src/rdb.c#L2784

So, `rdbLoad()` is blocking from caller's point of view. Loading RDB is non-blocking from client's point of view. 
Clients get `-LOADING` reply.

In our case, stacktrace becomes something like:

```c
processCommandFromLeader()
raft_recv_snapshot();
rdbLoad();
startProcessingAnotherCommandFromLeader();
assert(false);
```

We need to prevent processing the next message of the leader node at this point to avoid that assert. Blocking the connection will prevent it as far as I can see.

Reply:
@sjpotter Do you have any other comments? Otherwise, let's merge this as a workaround for now.